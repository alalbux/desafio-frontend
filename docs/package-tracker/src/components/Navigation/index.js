import React, { useState } from 'react';
import {
  IconButton,
  Menu,
  MenuItem,
  Button,
  Box,
  Paper,
  useMediaQuery,
} from '@mui/material';
import MenuIcon from '@mui/icons-material/Menu';
import MenuMobile from '@/components/MenuMobile';
import LOCALES from '@/locales/navigation';

const { MENU } = LOCALES;

const Navigation = () => {
  const [isDrawerOpen, setIsDrawerOpen] = useState(false);
  const [anchorEl, setAnchorEl] = useState(null);
  const [submenuOpen, setSubmenuOpen] = useState(null);

  const isMobile = useMediaQuery('(max-width:768px)');

  const toggleDrawer = (open) => () => {
    setIsDrawerOpen(open);
  };

  const openMobileMenu = (event) => {
    setAnchorEl(event.currentTarget);
    toggleDrawer(true);
  };

  const closeMobileMenu = () => {
    setAnchorEl(null);
  };

  const handleSubmenuOpen = (subItems) => {
    setSubmenuOpen(subItems);
  };

  const handleSubmenuClose = () => {
    setSubmenuOpen(null);
  };

  const renderMobileMenu = () => {
    return (
      <>
        <IconButton
          edge="start"
          color="inherit"
          aria-label="menu"
          onClick={openMobileMenu}
        >
          <MenuIcon />
        </IconButton>
        <Menu
          anchorEl={anchorEl}
          open={Boolean(anchorEl)}
          onClose={closeMobileMenu}
        >
          {MENU.map((menuItem, indexMobile) => (
            <MenuItem
              key={indexMobile}
              onClick={() => handleSubmenuOpen(menuItem.submenu)}
              sx={styles.mobileMenuItem}
            >
              {menuItem.label}
            </MenuItem>
          ))}
        </Menu>
      </>
    );
  };

  const renderDesktopMenu = () => {
    return (
      <>
        {MENU.map((menuItem, index) => (
          <div
            key={index}
            sx={styles.desktopMenuItem}
            onMouseEnter={() => handleSubmenuOpen(menuItem.submenu)}
            onMouseLeave={handleSubmenuClose}
          >
            <Button variant="text">{menuItem.label}</Button>
            {submenuOpen === menuItem.submenu &&
              menuItem.submenu.length > 0 && (
                <Paper sx={styles.submenu}>
                  {menuItem.submenu.map((subItem, subIndex) => (
                    <MenuItem
                      key={subIndex}
                      component="a"
                      href={subItem.redirection}
                    >
                      {subItem.label}
                    </MenuItem>
                  ))}
                </Paper>
              )}
          </div>
        ))}
      </>
    );
  };

  return (
    <div>
      <Box sx={styles.container}>
        {isMobile ? renderMobileMenu() : renderDesktopMenu()}
      </Box>

      <MenuMobile
        isDrawerOpen={isDrawerOpen}
        toggleDrawer={toggleDrawer}
        isOpen={isDrawerOpen}
      />
    </div>
  );
};

const styles = {
  container: {
    display: 'flex',
    flexDirection: 'row-reverse',
    alignItems: 'center',
  },
  mobileMenuItem: {
    display: 'block',
    '&:hover': {
      backgroundColor: '#FCFCFC',
    },
  },
  desktopMenuItem: {
    position: 'relative',
    display: 'block',
    '&:hover': {
      backgroundColor: '#FCFCFC',
    },
  },
  submenu: {
    position: 'absolute',
    top: 50,
    zIndex: 999,
    backgroundColor: '#FFFFFF',
    borderRadius: 4,
    display: 'flex',
    flexDirection: 'column',
    padding: 2,
  },
};

export default Navigation;
