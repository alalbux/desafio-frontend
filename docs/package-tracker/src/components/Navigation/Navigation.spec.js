import React from 'react';
import { render } from '@testing-library/react';
import Navigation from '@/components/Navigation';

describe('Navigation component', () => {
  it('matches the snapshot', () => {
    const { container } = render(<Navigation />);
    expect(container).toMatchSnapshot();
  });
});
