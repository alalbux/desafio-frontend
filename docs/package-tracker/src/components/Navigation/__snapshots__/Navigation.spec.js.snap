// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component matches the snapshot 1`] = `
<div>
  <div>
    <div
      class="MuiBox-root css-1mfpmt0"
    >
      <div
        sx="[object Object]"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Para vocÃª
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        sx="[object Object]"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Para empresas
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        sx="[object Object]"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Para entregar
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        sx="[object Object]"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Venda e suporte
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        sx="[object Object]"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Acompanha seu pedido
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
